agent/config/config.go

```diff
@@ -148,6 +148,7 @@ func DefaultConfig() Config {
 		DockerGraphPath:         "/var/lib/docker",
 		ReservedMemory:          0,
 		AvailableLoggingDrivers: []dockerclient.LoggingDriver{dockerclient.JsonFileDriver},
+		PrivilegedCapable:       true,
```

Comment:
Setting this to `true` means that it will always get merged in.

The way merging is done here is that if a value is its zero value (`false` for `bool`), the value is considered unchanged.
In order for this to work, you'll need to change this to be a `*bool` type instead.